
     This patch exports the RPC symbols and headers, as the previous glibc
     version do.  When libtirpc works as a proper and full replacement for RPC,
     then use libtirpc.

     This is shamelessly nicked from LFS which seems to have taken from Fedora.

     Douglas Jerome <douglas@ttylinux.org>

diff -Naur glibc-2.14/include/libc-symbols.h glibc-2.14/include/libc-symbols.h
--- glibc-2.14/include/libc-symbols.h	2011-05-30 21:12:33.000000000 -0700
+++ glibc-2.14/include/libc-symbols.h	2012-04-06 21:30:48.261001171 -0700
@@ -635,7 +635,7 @@
 # define libc_hidden_proto(name, attrs...) hidden_proto (name, ##attrs)
 # define libc_hidden_def(name) hidden_def (name)
 # define libc_hidden_weak(name) hidden_weak (name)
-# define libc_hidden_nolink(name, version) hidden_nolink (name, libc, version)
+# define libc_hidden_nolink(name, version) hidden_def (name)
 # define libc_hidden_ver(local, name) hidden_ver (local, name)
 # define libc_hidden_data_def(name) hidden_data_def (name)
 # define libc_hidden_data_weak(name) hidden_data_weak (name)
diff -Naur glibc-2.14/nis/Makefile glibc-2.14/nis/Makefile
--- glibc-2.14/nis/Makefile	2011-05-30 21:12:33.000000000 -0700
+++ glibc-2.14/nis/Makefile	2012-04-06 21:32:27.353762342 -0700
@@ -23,9 +23,9 @@
 
 aux			:= nis_hash
 
+headers			:= $(wildcard rpcsvc/*.[hx])
 distribute		:= nss-nis.h nss-nisplus.h nis_intern.h Banner \
-			   nisplus-parser.h nis_xdr.h nss \
-			   $(wildcard rpcsvc/*.[hx])
+			   nisplus-parser.h nis_xdr.h nss
 
 # These are the databases available for the nis (and perhaps later nisplus)
 # service.  This must be a superset of the services in nss.
diff -Naur glibc-2.14/sunrpc/Makefile glibc-2.14/sunrpc/Makefile
--- glibc-2.14/sunrpc/Makefile	2011-05-30 21:12:33.000000000 -0700
+++ glibc-2.14/sunrpc/Makefile	2012-04-06 21:37:58.105627375 -0700
@@ -53,7 +53,7 @@
 				    des_crypt.h)
 headers-not-in-tirpc = $(addprefix rpc/,key_prot.h rpc_des.h) \
 		       $(rpcsvc:%=rpcsvc/%) rpcsvc/bootparam.h
-headers = rpc/netdb.h
+headers = rpc/netdb.h $(headers-in-tirpc) $(headers-not-in-tirpc)
 install-others = $(inst_sysconfdir)/rpc
 generated = $(rpcsvc:%.x=rpcsvc/%.h) $(rpcsvc:%.x=x%.c) $(rpcsvc:%.x=x%.stmp) \
 	    $(rpcsvc:%.x=rpcsvc/%.stmp) rpcgen
